sdk/dapp-kit/src/hooks/wallet/useConnectWallet.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UseMutationOptions } from '@tanstack/react-query';
+import { useMutation } from '@tanstack/react-query';
+import type {
+	StandardConnectInput,
+	StandardConnectOutput,
+	WalletAccount,
+} from '@mysten/wallet-standard';
+import { useWalletContext } from '../../components/WalletProvider.js';
+import { WalletAlreadyConnectedError, WalletNotFoundError } from '../../errors/walletErrors.js';
+import {
+	getMostRecentWalletConnectionInfo,
+	setMostRecentWalletConnectionInfo,
+} from '../../utils/walletUtils.js';
+import { walletMutationKeys } from '../../constants/walletMutationKeys.js';
+
+type ConnectWalletArgs = {
+	/** The name of the wallet as defined by the wallet standard to connect to. */
+	walletName: string;
```

Comment:
Thoughts on passing the name vs the entire wallet object?

Reply:
I could argue both ways 😅

We can always grab the wallet object from the wallet name since wallet names are unique, so that's nice. It kind of comes down to the better user interface. Off the top of my head, I think just requiring the wallet name is easier for consumers in some scenarios:

```
const {wallets} = useWallets();
const ethosWallet = wallets.find(wallet => wallet.name === 'Ethos Wallet');
...
connectWallet({ wallet: ethosWallet });
```

vs.

```
connectWallet({ wallet: 'Ethos Wallet' });
```


```
  
```

Reply:
Argument for the former: You shouldn't blindly connect to wallet names since it might not be present in the wallet list

Reply:
Yeah touche - I forgot the conditional checking part when writing that example lol. Let's go with specifying the whole wallet object since that adds some enforceability then?

Reply:
Yeah I'm happy with that. We can always change this too